html/js/tree/Node.js

```diff
@@ -0,0 +1,118 @@
+GAMBIT.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new Node.
+    * @class
+    * @param {Node} [parent] Parent node. If null, this is root.
+    */
+    function Node(parent) {
+        this.parent = parent;
+        this.children = [];
+
+        if (parent === null) { // If this is root set level to 0
+            this.level = 0;
+        } else {
+            parent.addChild(this);
+            this.level = parent.level + 1;
+        }
+    }
+
+    /**
+    * ToString function
+    */
+    Node.prototype.toString = function nodeToString() {
+        return "Node: " + "children.length: " + this.children.length + "; level: " + this.level;
+    };
+
+    /**
+    * Function that draws the node in the global canvas
+    */
+    Node.prototype.draw = function () {
+        var thisNode = this;
+        var circle = GAMBIT.canvas.circle(GAMBIT.CONSTANTS.CIRCLE_SIZE)
+            .addClass('node')
+            .x(this.x)
+            .y(this.y)
+            .click(function() {
```

Comment:
Actually no, this may have issues with "this" and may not work, so leave it as is